src/Printer6.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace mheap\GithubActionsReporter;
+
+use PHPUnit\TextUI\ResultPrinter;
+use PHPUnit\Framework\TestFailure;
+use PHPUnit\Framework\TestResult;
+
+use function mheap\GithubActionsReporter\Functions\printDefectTrace;
+
+class Printer6 extends ResultPrinter
+{
+    /**
+     * @var null|string
+     */
+    private $currentType;
+
+    protected function printHeader(): void
+    {
+    }
+
+    protected function writeProgress($progress): void
+    {
+    }
+
+    protected function printFooter(TestResult $result): void
+    {
+    }
+
+    protected function printDefects(array $defects, $type): void
+    {
+        $this->currentType = (in_array($type, ['error', 'failure']) === true) ? 'error' : 'warning';
```

Comment:
This is repeated in all printers - should it be a helper function?

Reply:
It would have been ideally, and originally I had done so.
But because there is state required (ie. `$this->printDefect()` and `$this->currentType`) it's not viable to do